### PR TITLE
Refactor PCM code (chunk 1)

### PR DESCRIPTION
**Pull request by ErikPrantare**
First chunk of refactoring for PCM, more to come.

Please tell if you would prefer larger or smaller PRs for these kinds of
changes.

### CODE REVIEW

**Comment by ErikPrantare** [comment]
Seems like many of the changes from #594 ended up showing here too, changes are from [5512e96: Make values constexpr where possible](https://github.com/projectM-visualizer/projectm/pull/595/commits/5512e965af8868e892f3b99777decd9e6e5772a0) onwards.

**Comment by kblaschke** [comment]
> Seems like many of the changes from #594 ended up showing here too

Sure, as the PR shows all differences between your branch and the target, so it includes all commits from your other PR, as these commits are part of this branch, too. These will vanish though as soon as the other PR is merged. Also don't forget to rebase this branch again onto the other one  in #594 after the squash, so there won't be duplicated commits.

**Comment by ErikPrantare** [comment]
Will do, and will make sure to organize my branches better in the future.

**Comment by kblaschke** [comment]
Can always get messy with multiple PRs open, sometimes there's simply no "good" way. Never be afraid to rebase and force-push, as long as the branch isn't merged, you can change everything at any time.

**Comment by ErikPrantare** [comment]
I think I've gotten the history correct now

File: src/libprojectM/PCM.cpp [hunk 1]
```diff
@@ -125,7 +125,7 @@ void PCM::addPCMfloat(const float* PCMdata, size_t samples)
         size_t j=(i+start)%maxsamples;
         a=pcmL[j] = PCMdata[i];
         pcmR[j] = PCMdata[i];
-        sum += fabs(a);
+        sum += std::abs(a);
```
Comment (kblaschke) [inline src/libprojectM/PCM.cpp:85]: Suggest placing a `using std::abs;` at the top of the implementation file to make the calls a bit less verbose (same for `std::max`).

Comment (ErikPrantare) [reply src/libprojectM/PCM.cpp:85]: I'm not a fan of file-wide using declarations when it comes to libraries, but if you think it'd be more readable maybe they can be put locally? Always `std::abs` would be consistent, if we have exceptions to that I think we should indicate that as locally as possible. Personally my brain just filters out the `std::`, like lipsers filter out all the parentheses when they code :^)

Comment (kblaschke) [reply src/libprojectM/PCM.cpp:85]: Can just keep it that way. Would be more of an issue with a class in a nested namespace, like the Poco library often has, but in this case `using namespace` is a good solution if these classes are heavily used.

File: src/libprojectM/PCM.cpp [hunk 2]
```diff
@@ -24,380 +24,280 @@
...
+                  "rdft invariant not preserved: n is power of two");
+    rdft(aSize, isgn, a.data(), ip.data(), w.data());
 }
 
 
 // puts sound data requested at provided pointer
 //
 // samples is number of PCM samples to return
-// smoothing is the smoothing coefficient
+// smoothing does nothing
```
Comment (kblaschke) [inline src/libprojectM/PCM.cpp:210]: Exactly, this should be removed.

**Review by kblaschke — APPROVED** [review]

**Pull request merged**